### PR TITLE
Less hardcoded Scala versions in tests and some bumps

### PR DESCRIPTION
**Pull request by lefou**
- Moved some test Scala version definitions up (into build.sc when possible)
- Removed some hardcoded scala versions
- Avoid compilation with Scala 2.12.4 compile, which is known to have issues with newer zinc (fixes some already present warnings)

### CODE REVIEW

File: contrib/scoverage/test/src/HelloWorldTests.scala [hunk 1]
```diff
@@ -190,14 +190,14 @@ trait HelloWorldTests extends utest.TestSuite {
...
+  override def testScoverageVersion = sys.props.getOrElse("MILL_SCOVERAGE_VERSION", ???)
   override def testScalatestVersion = "3.0.8"
 }
 
 object HelloWorldTests_2_13 extends HelloWorldTests {
   override def threadCount = Some(1)
-  override def testScalaVersion: String = sys.props("MILL_SCALA_2_13_VERSION")
-  override def testScoverageVersion = sys.props("MILL_SCOVERAGE_VERSION")
+  override def testScalaVersion: String = sys.props.getOrElse("TEST_SCALA_2_13_VERSION", ???)
+  override def testScoverageVersion = sys.props.getOrElse("MILL_SCOVERAGE_VERSION", ???)
```
Comment (lolgab) [inline contrib/scoverage/test/src/HelloWorldTests.scala:201]: Why this change?
```scala
@ Map(1 -> 1)(2)
java.util.NoSuchElementException: key not found: 2
  scala.collection.immutable.Map$Map1.apply(Map.scala:245)
  ammonite.$sess.cmd0$.<clinit>(cmd0.sc:1)


@ Map(1 -> 1).getOrElse(2, ???)
scala.NotImplementedError: an implementation is missing
  scala.Predef$.$qmark$qmark$qmark(Predef.scala:344)
  ammonite.$sess.cmd1$.$anonfun$res1$1(cmd1.sc:1)
  scala.collection.immutable.Map$Map1.getOrElse(Map.scala:250)
  ammonite.$sess.cmd1$.<clinit>(cmd1.sc:1)
```

**Review by lolgab — COMMENTED** [review]
I need to leave a main comment to leave a comment in code. Sorry.

File: contrib/scoverage/test/src/HelloWorldTests.scala (hunk 1 above)
Comment (lefou) [reply contrib/scoverage/test/src/HelloWorldTests.scala:201]: `sys.props` has a `def default(key: String) = null` override.

**Comment by lefou** [comment]
> I need to leave a main comment to leave a comment in code. Sorry.

No problem, but I don't see this restriction with my browser.

Any idea what's wrong with the NodeJS tests on Java 11?

File: contrib/scoverage/test/src/HelloWorldTests.scala (hunk 1 above)
Comment (lolgab) [reply contrib/scoverage/test/src/HelloWorldTests.scala:201]: 🤯

**Comment by lefou** [comment]
Looks like it's combination of Scala 2.12.5 and Scala.js 0.6.3{2,3} which doesn't work.

**Comment by lefou** [comment]
@lolgab This PR is a preparation for Zinc version bump.

**Review by lolgab — APPROVED** [review]
